app/kubemci/pkg/gcp/urlmap/urlmapsyncer.go

```diff
@@ -142,11 +163,16 @@ func (s *URLMapSyncer) desiredURLMap(lbName string, ing *v1beta1.Ingress, beMap
 	}
 	gceMap, err := s.ingToURLMap(ing, beMap)
 	if err != nil {
+		fmt.Println("Error getting URL map from Ingress:", err)
 		return nil, err
 	}
 	um.DefaultService = gceMap.GetDefaultBackend().SelfLink
-	um.HostRules = []*compute.HostRule{}
-	um.PathMatchers = []*compute.PathMatcher{}
+	if len(gceMap) > 0 {
```

Comment:
I added this length check to fix the DeepEqual problem I was having- it was an empty slice in "desired" and just "nil" from the server.